drafts/draft-0005.md

```diff
@@ -0,0 +1,188 @@
+}
+```
+
+####  CORS headers
+
+In order to comply with browser cross-origin access policies, the service should provide wildcard CORS response HTTP header. The following HTTP header must be set for the API endpoints:
+
+```
+Access-Control-Allow-Origin: *
+```
+
+### Multisig Transaction Flow
+
+1. Determine whether a transaction requires additional signatures in order to be valid.
+1. If the transaction requires more signatures, look up the data entry with the key `multisig.domain` on the transaction source account.
+1. Resolve the `stellar.toml` associated with the account's `multisig.domain` entry and return the `MULTISIG_ENDPOINT` field.
+1. Convert the transaction to a SEP-0007 style URI, setting any additional parameters such as `callback`.
+1. Before submitting the transaction, it should contain at least one valid signature.
+1. `POST` to the `MULTISIG_ENDPOINT` endpoint with the SEP-0007 `tx` URI in the body as the parameter `uri`.
+1. The service responds with a 200 HTTP status code and a JSON object specified in the Submit Transaction Endpoint Response section.
+
+## Rationale
+
+When implementing StellarGuard for the Stellar Account Viewer, it was rightly pointed out it should avoid vendor-specific implementation. This SEP provides an alternative approach that can be used by tools that want to support multisig accounts without implementing a multisig server for each product. 
+
+One design choice was whether to skip the `stellar.toml` step and add the fully qualified multisig server endpoint to the account's data field. The `stellar.toml` approach affords greater flexibilty for the server to change its implementations. It is much easier to update the `stellar.toml` than to update the data entry of every account that uses it.
+
+## Implementations
+
+The server implementation is currently live on [https://test.stellarguard.me](https://test.stellarguard.me) and [https://stellarguard.me](https://stellarguard.me).
+
+A library that implements this protocol has been published here: [https://github.com/stellarguard/multisig-utils](https://github.com/stellarguard/multisig-utils)
+
+A demo using that library that submits to the StellarGuard multisig endpoint: [https://stellarguard.github.io/multisig-utils/demo/](https://stellarguard.github.io/multisig-utils/demo/)
+
+## Outstanding Questions/Discussion
+
+1. Should we specify that multisig services expose and return a `statusHref` field for status updates in the response data and also dictate how that should endpoint should function? Applications may have specific use-cases or implementations that this may not work with (for instance, what if they want to use event source or websockets instead of a polling endpoint -- or if they do not want to expose a public status endpoint at all for privacy reasons). However, if we do not specify exactly how it should work then it's possible there will be multiple incompatible implementations.
+
```

Comment:
* If we use txHashes we can remove the whole statusHref thing.
* Dictate how the endpoint should function is the job of this SEP.
* We should allow vendor-specific extension, and let them out of the realm of this SEP, or, we can make event source / websocket an *optional* extension whose behavior is specified by this SEP. In any case coordinators/utils must implement the base protocol.

Reply:
There are already people implementing that so I think it would be nicer to work on that now rather than saying them to do it again because it's not standard later. Also wallet are going to need to know how they are supposed to handle listening. I don't think it should be a mandatory thing but that's needed.

Reply:
When you say “wallets are going to need to know how they are supposed to handle listening”, what are you referring to? What will wallets listen for? — I fear that this conversation is going to go down the thread of multisig coordination so I will say it again, this SEP is **not a protocol for how to do multisig coordination**.

Reply:
Resolved